### ISSUE TITLE
[✨] Switch from Eslint to Biome

### ISSUE THREAD

**Issue by rogervila** [QwikDev/qwik#6648]
### Is your feature request related to a problem?

This feature is unrelated to any problem but represents a significant performance improvement for the build step since Biome linter is much faster than Eslint.

Biome provides a [migration tool](https://biomejs.dev/guides/migrate-eslint-prettier/) that makes it a breeze to migrate the Eslint configuration.

### Describe the solution you'd like

The solution would consist of a PR with the following changes:

- [ ] Migrate Eslint configuration to Biome, including rules defined in `eslint-plugin-qwik`, using Biome's [migration tool](https://biomejs.dev/guides/migrate-eslint-prettier/) 
- [ ] Install Biome and uninstall Eslint-related packages from `package.json`
- [ ] Modify the `lint` command to use Biome instead of Eslint
- [ ] Deprecate `eslint-plugin-qwik`

### Describe alternatives you've considered

AFAIK, Biome is the most performant linter on the market and, since it provides a [migration tool](https://biomejs.dev/guides/migrate-eslint-prettier/), I have not considered other alternatives.

### Additional context

Performance comparison between Eslint and Biome published on [Biome's website](https://biomejs.dev/):

![eslint-vs-biome](https://github.com/QwikDev/qwik/assets/6053012/a0258d9c-6b49-4efd-b9ef-bdc19d9b4c66)

**Comment by rogervila** [comment]
If the request gets enough support I will start the previously mentioned PR

**Comment by PatrickJS** [comment]
yes, I want to switch to Biome asap the only concerns we have are tailwind prettier plugin and qwik eslint plugin that both need to work with Biome. If there is now a path and you're willing to work with us on a PR then I don't see why not

**Comment by thejackshelton** [comment]
Hey @rogervila! This would be amazing. Would love to help out on this.

**Comment by PatrickJS** [comment]
@rogervila can you ping us in the discord so we can create a channel for this effort

**Comment by thejackshelton** [comment]
Hey @rogervila! Any status updates? Would love to help you on this effort.

**Comment by shirotech** [comment]
+1 Looking forward to this being shipped

**Comment by rogervila** [comment]
Hi @thejackshelton @shirotech I will start working on it next week together with @PatrickJS :)

**Comment by maiieul** [comment]
I'd absolutely love to use Biome instead of Eslint/prettier. But we'll need to make sure current users of `eslint-plugin-qwik` and eslint v8 aren't forced to use Biome. They might not be able to migrate if Biome isn't compatible with other tools they use.
